### PR TITLE
Add medically needy pathway for Medicaid

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
MedicaidScreener had some helpful explanations here - same sort of implementation using the KFF table [here](https://files.kff.org/attachment/Issue-Brief-Medicaid-Financial-Eligibility-for-Seniors-and-People-with-Disabilities-Findings-from-a-50-State-Survey).

### CODE REVIEW

File: openfisca_us/parameters/hhs/medicaid/eligibility/categories/medically_needy/limit/assets/couple.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
+description: Maximum assets for a couple to qualify for Medicaid under the 'medically needy' pathway.
+metadata:
+  unit: currency-USD
+  period: month
+  name: medicaid_medically_needy_assets_couple
+  label: Medicaid medically needy pathway for couples asset limit
+
```
Comment (nikhilwoodruff) [inline openfisca_us/parameters/hhs/medicaid/eligibility/categories/medically_needy/limit/assets/couple.yaml:7]: Add references

File: openfisca_us/variables/gov/hhs/medicaid/eligibility/categories/medically_needy/is_medically_needy_for_medicaid.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            is_joint,
+            mn.limit.income.couple[state] * MONTHS_IN_YEAR,
+            mn.limit.income.individual[state] * MONTHS_IN_YEAR,
+        )
+        asset_limit = where(
+            is_joint,
+            mn.limit.assets.couple[state],
+            mn.limit.assets.individual[state],
+        )
+        under_limits = (income < income_limit) & (assets < asset_limit)
```
Comment (MaxGhenis) [inline openfisca_us/variables/gov/hhs/medicaid/eligibility/categories/medically_needy/is_medically_needy_for_medicaid.py:32]: Confirming not `<=`?

**Review by MaxGhenis — APPROVED** [review]

File: openfisca_us/variables/gov/hhs/medicaid/eligibility/categories/medically_needy/is_medically_needy_for_medicaid.py (hunk 2 above)
Comment (nikhilwoodruff) [reply openfisca_us/variables/gov/hhs/medicaid/eligibility/categories/medically_needy/is_medically_needy_for_medicaid.py:32]: It's <= per the CFR

File: openfisca_us/variables/gov/hhs/medicaid/eligibility/categories/medically_needy/is_in_medicaid_medically_needy_category.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+class is_in_medicaid_medically_needy_category(Variable):
+    value_type = bool
+    entity = Person
+    label = "In Medicaid medically needy category"
+    documentation = "Whether this person is in a Medicaid category for which there is a medically needy pathway."
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        ma = parameters(
```
[the remark this answers is not in the document]
Comment (nikhilwoodruff) [reply openfisca_us/variables/gov/hhs/medicaid/eligibility/categories/medically_needy/is_in_medicaid_medically_needy_category.py]: Renamed

**Pull request merged**